### DISCUSSION TITLE
How to test fluentui v9 combobox correctly?

### DISCUSSION THREAD

**Discussion by rosergeev** [microsoft/fluentui#30821] [General] [answered]
Hi there!
I don't get how to write tests that are related to combobox or other fluentui components that dynamically insert something to the root node of the DOM, e.g. Calendar's popup.
Here it is the simple test repro https://codesandbox.io/p/devbox/testcombobox-jyvp8t?file=%2Fsrc%2FComponents%2FComboWrapper.test.tsx%3A6%2C49
In this test I would like programmatically open Combobox and select second element.
How to achieve this?

**Comment by bsunderhus** [comment, marked as answer]
The `listbox` (popover) of a `Combobox` is inside a `Portal`, and portals do not render on node environment. I'd recommend using some browser environment testing tool, like [playwright](https://playwright.dev/) or [cypress](https://www.cypress.io/), as in those environments portals would render.

**Reply by rosergeev** [reply]
Thank you for your answer, now it's much clearer!
